pallenec

```diff
@@ -63,15 +64,43 @@ local inspect_opts = {
     end
 }
 
+local function print_tokens(lexer)
+    while true do
+        local token = lexer:next()
+        if token.name == "EOF" then
+            break
+        end
+
+        local loc = token.loc
+        local end_loc = token.end_loc
+        local value
+        if token.value then
+            if type(token.value) == "string" then
+                value = " " .. C.string(token.value)
+            else
+                value = " " .. tostring(token.value)
+            end
+        else
+            value = ""
+        end
+        print(string.format("%d:%d-%d:%d %d-%d %s%s", loc.line, loc.col,
+            end_loc.line, end_loc.col, loc.pos, end_loc.pos, token.name, value))
+    end
+end
+
 local function dump(filename, stop_after)
     local input, err = driver.load_input(filename)
     if err then util.abort(err) end
 
     local out, errs = driver.compile_internal(filename, input, stop_after, opt_level)
-    if not out then util.abort(table.concat(errs, "\n")) end
+    if stop_after == "lexer" then
+        print_tokens(out)
+    else
+        if not out then util.abort(table.concat(errs, "\n")) end
```

Comment:
Should this error handling be right after the call to `compile_internal`, before the `if stop_after`?